R2API/ItemDrop/Catalog.cs

```diff
@@ -6,62 +6,72 @@ namespace R2API {
             public static readonly List<EquipmentIndex> EliteEquipment = new List<EquipmentIndex>();
-
             public static readonly List<ItemIndex> Pearls = new List<ItemIndex>();
+            public const string NullIconTextureName = "texNullIcon";
 
-            public static void PopulateItemCatalog() {
-                if (!Loaded) {
-                    Pearls.Add(ItemCatalog.FindItemIndex("Pearl"));
-                    Pearls.Add(ItemCatalog.FindItemIndex("ShinyPearl"));
-                    foreach (var itemIndex in ItemCatalog.allItems) {
-                        var itemDef = ItemCatalog.GetItemDef(itemIndex);
-                        if (itemDef.tier != ItemTier.NoTier && itemDef.pickupIconSprite != null && itemDef.pickupIconSprite.name != DropList.NullIconTextureName) {
-                            if (itemDef.ContainsTag(ItemTag.Scrap)) {
-                                if (!ScrapItems.ContainsKey(itemDef.tier)) {
-                                    ScrapItems.Add(itemDef.tier, itemIndex);
-                                }
-                            }
-                            else if (itemDef.ContainsTag(ItemTag.WorldUnique)) {
-                                SpecialItems.Add(itemIndex);
+            public static void PopulateCatalog() {
+                SpecialItems.Clear();
+                ScrapItems.Clear();
+                EliteEquipment.Clear();
+                Pearls.Clear();
+
+                Pearls.Add(ItemCatalog.FindItemIndex("Pearl"));
+                Pearls.Add(ItemCatalog.FindItemIndex("ShinyPearl"));
+                foreach (var itemIndex in ItemCatalog.allItems) {
+                    var itemDef = ItemCatalog.GetItemDef(itemIndex);
+                    if (itemDef.tier != ItemTier.NoTier && itemDef.pickupIconSprite != null && itemDef.pickupIconSprite.name != NullIconTextureName) {
+                        if (itemDef.ContainsTag(ItemTag.Scrap)) {
+                            if (!ScrapItems.ContainsKey(itemDef.tier)) {
+                                ScrapItems.Add(itemDef.tier, itemIndex);
                             }
+                        } else if (itemDef.ContainsTag(ItemTag.WorldUnique)) {
+                            SpecialItems.Add(itemIndex);
                         }
                     }
+                }
                     
-                    foreach (var equipmentIndex in EquipmentCatalog.allEquipment) {
-                        var equipmentDef = EquipmentCatalog.GetEquipmentDef(equipmentIndex);
-                        if (!EquipmentCatalog.equipmentList.Contains(equipmentIndex)) {
-                            if (equipmentDef.pickupIconSprite != null &&
-                                equipmentDef.pickupIconSprite.name != DropList.NullIconTextureName) {
-                                EliteEquipment.Add(equipmentIndex);
-                            }
+                foreach (var equipmentIndex in EquipmentCatalog.allEquipment) {
```

Comment:
You can find this in the elitecatalog too. That might be more reliable if you're trying to find elite equipment than it would be to iterate through this like this. If an elite has an equipment associated with them (and all I've seen do) it's under that elitedef's equipmentdef field.